kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/commands/application/StorageAwareApplicationCommandRegistry.kt

```diff
@@ -0,0 +1,135 @@
+@file:Suppress(
+    "UNCHECKED_CAST"
+)
+@file:OptIn(
+    ExperimentalCoroutinesApi::class
+)
+
+package com.kotlindiscord.kord.extensions.commands.application
+
+import com.kotlindiscord.kord.extensions.commands.application.message.MessageCommand
+import com.kotlindiscord.kord.extensions.commands.application.slash.SlashCommand
+import com.kotlindiscord.kord.extensions.commands.application.user.UserCommand
+import com.kotlindiscord.kord.extensions.registry.RegistryStorage
+import dev.kord.common.entity.Snowflake
+import dev.kord.core.event.interaction.ChatInputCommandInteractionCreateEvent
+import dev.kord.core.event.interaction.MessageCommandInteractionCreateEvent
+import dev.kord.core.event.interaction.UserCommandInteractionCreateEvent
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.firstOrNull
+import kotlinx.coroutines.flow.toList
+
+/**
+ * ApplicationCommandRegistry which acts based off a specified storage interface.
+ *
+ * Discord lifecycles may not be implemented in this class and require manual updating.
+ */
+public class StorageAwareApplicationCommandRegistry(
```

Comment:
I feel like it'd probably be useful for some people if this was `open` and also had `public` access to everything you've restricted access to below - otherwise people will have to copy-paste the code in this class if they want to build upon it.

Reply:
Sure, this should be an easy fix.